FirebaseRemoteConfig/Sources/FIRRemoteConfig.m

```diff
@@ -214,8 +214,11 @@ - (void)ensureInitializedWithCompletionHandler:
 #pragma mark - fetch
 
 - (void)fetchWithCompletionHandler:(FIRRemoteConfigFetchCompletion)completionHandler {
-  [self fetchWithExpirationDuration:_settings.minimumFetchInterval
-                  completionHandler:completionHandler];
+  __block NSTimeInterval minimumFetchInterval;
+  dispatch_sync(_queue, ^{
```

Comment:
In general `dispatch_sync` should be avoided unless required because the more `dispatch_sync` is used the easier it is to introduce a deadlock even when modifying code not directly related. I wonder if it can be updated to either:
1. Use `atomic` properties or `_Atomic` vars for variables involved in the race condition
2. Use `dispatch_async` e.g. : 
```
dispatch_async(_queue, ^{
  [self fetchWithExpirationDuration:self->_settings.minimumFetchInterval completionHandler:completionHandler];
}
```

Reply:
#2 is probably better as remote config's mutex lock is all guarded by this queue. The fetch is assigned to async again inside https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseRemoteConfig/Sources/RCNFetch.m#L139
So using another mechanism might be more expensive since we already have a queue just for that.